build/Dockerfile

```diff
@@ -1,9 +1,9 @@
-FROM quay.io/app-sre/boilerplate:image-v5.0.1 AS builder
+FROM registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.22-openshift-4.17 AS builder
```

Comment:
Do you think we can use this one? My feeling is rather not, because then it depends on the registry.ci.openshift.org and I guess we don't want that or?

Reply:
I wanted to get a generic image. This one was used also on boilerplate, so I considered appropriate for us as well. In the end we only need an image based on rhel 8 with golang pre-installed (or doing it manually).

In the end is not that different, we could stick with boilerplate image as well, as we won't use any more of their makefiles.

Reply:
I am not sure...just note that the access (user, pwd) to the original quay repo are different than the `registry.ci.openshift.org` so not sure if this can cause some issues during building.

Reply:
rolling it back, as agreed